src/cc/libbpf.c

```diff
@@ -669,7 +669,7 @@ static int libbpf_bpf_prog_load(const struct bpf_load_program_attr *load_attr,
   p.line_info = load_attr->line_info;
   p.prog_flags = load_attr->prog_flags;
 
-  return bpf_prog_load(load_attr->prog_type, load_attr->name, load_attr->license,
+  return bpf_prog_load_v0_6_0(load_attr->prog_type, load_attr->name, load_attr->license,
```

Comment:
bpf_prog_load_v0_6_0 will go away in libbpf 1.0, so we cannot use bpf_prog_load_v0_6_0 here.
Could you explain what exactly bpf_prog_load caused the problem?

Reply:
`libbpf_bpf_prog_load()` call `bpf_prog_load()` with:

```
 670   p.prog_flags = load_attr->prog_flags;
 671 
 672   return bpf_prog_load(load_attr->prog_type, load_attr->name, load_attr->license,
 673                        load_attr->insns, load_attr->insns_cnt, &p);
 674 }
```

however

```
#5  0x00007ffff7f7d917 in bpf_prog_load_xattr2 (attr=attr@entry=0x7fffffffaf90, pobj=0x5555557a310b, prog_fd=0x7fffffffc590) at libbpf.c:10093
#6  0x00007ffff7f7dad3 in bpf_prog_load_deprecated (file=<optimized out>, type=<optimized out>, pobj=<optimized out>, prog_fd=<optimized out>) at libbpf.c:10158
#7  0x00007ffff1025c78 in libbpf_bpf_prog_load (load_attr=0x7fffffffb210, log_buf=0x0, log_buf_sz=0) at /home/rongtao/Git/bcc/src/cc/libbpf.c:672
#8  0x00007ffff10260af in bcc_prog_load_xattr (attr=0x7fffffffb210, prog_len=16, log_buf=0x7fffffffb540 "", log_buf_size=4096, allow_rlimit=true)
    at /home/rongtao/Git/bcc/src/cc/libbpf.c:758
```

`frame 6` in libbpf

```
1180 LIBBPF_API int bpf_prog_load_deprecated(const char *file, enum bpf_prog_type type,                                                     
1181                     struct bpf_object **pobj, int *prog_fd);
```

when force `const char *file = load_attr->prog_type`, SEGV happen.